validator/main.py

```diff
@@ -125,8 +168,23 @@ def call_llm(self, prompt: str) -> str:
             response (str): String representing the LLM response.
         """
         return self._llm_callable(prompt)
-
-    def evaluate_with_llm(self, text: str, query_function: Callable) -> str:
+    
+    def evaluate_with_vectara(self, text:str) -> bool:
+        classifier = pipeline(
+            "text-classification",
+            model="vectara/hallucination_evaluation_model",
+            tokenizer=AutoTokenizer.from_pretrained("google/flan-t5-base"),
+            trust_remote_code=True,
+            device="cpu" if torch.cuda.is_available() else "cpu",
+        )
+        result = classifier(text, batch_size=1)
+        if result[0]['label'] == 'consistent':
+            return True
+        if result[0]['label'] == 'hallucinated':
+            return False
```

Comment:
If neither condition is met, None is returned, which doesn't match into the supported/unsupported sentences append-ing on lines 224-227